model/wrapper.js

```diff
@@ -0,0 +1,44 @@
+var bcrypt = require('bcrypt')
+const saltRounds = 10
+
+function createUser(connection, uname, email, password, admin) {
+    bcrypt.hash(password, saltRounds, function (err, hash) {
+        var query = 'INSERT INTO users VALUES(' + email + ', ' + hash + ', ' + uname +  ', ' + admin + ')'
+        connection.query(query, function (err, rows, fields) {
+            if (err) {
+                throw "Failed to execute create user SQL"
+            } else {
+                return true
+            }
+        })
+    })
+}
+
+function userExists(connection, email) {
+    var query = 'SELECT uname FROM users WHERE email = ' + email
+    connection.query(query, function (err, rows, fields) {
+        if (err) {
+            throw "Failed to execute user exists SQL"
+        }
+        return (rows.count > 0)
+    })
+}
+
+function shouldLogin(connection, email, password) {
+    var query = 'SELECT passw FROM users WHERE email = ' + email
+    connection.query(query, function (err, rows, fields) {
+        if (err) {
+            throw "Failed to execute login check SQL"
+        }
+        if (rows.length > 0) {
```

Comment:
Something to keep in mind: it is possible to use a timing attack to determine whether the user exists in the database or not. This is because if the user exists, the bcrypt.compare takes longer to complete. This is just a nitpick though, probably not a big deal.

Reply:
ok well if you tell me what to change, I'll change it